### ISSUE TITLE
Each amonagent 'checkin' registering as unique server

### ISSUE THREAD

**Issue by hailthemelody** [amonapp/amon-issues#22]
Hi,

It appears that every 'checkin' from amonagent to the Amon service (I'm using Amon Personal currently to test it out) is registering as a unique server. I have experimented with fresh Amon Personal installs with and without SSL (changing the API keys on hosts running amonagent obviously), as well as with several different hosts running amonagent. Same result.

I am running Amon Personal on Ubuntu 14.04, installed via the `bash -c "$(curl -L https://www.amon.cx/install-personal/)"` command provided via https://www.amon.cx/docs/personal/. The amonagent checks came in from Debian 7 and Ubuntu 14.04 hosts.

![amon-personal](https://cloud.githubusercontent.com/assets/6835016/15292179/61357e30-1b71-11e6-9da7-c273909569f8.png)

_The above screenshot is of a recent experiment with only one host, though as I mentioned above the same issue occurred with more than one host._

**Comment by martinrusev** [comment]
@hailthemelody I will take a look. This is almost certainly an issue with the latest amonagent release. Specifically this line right here: https://github.com/amonapp/amonagent/blob/master/collectors/host.go#L150 

I do have a test suite for that, but for some reason it still fails. Maybe permissions issues. Will let you know soon.

**Comment by hailthemelody** [comment]
Thanks @martinrusev . Let me know if you'd like me to experiment with anything.

**Comment by hailthemelody** [comment]
If it's of any help, if I try to delete any of the 'servers' I get an Server Error (500). Screenshot:

![screenshot - 05172016 - 07 21 39 am](https://cloud.githubusercontent.com/assets/6835016/15320722/113449fe-1c00-11e6-9a24-c335c4af2268.png)

Relevant output from /var/log/amon/amonapp.log

`
[17/May/2016 07:24:34] ERROR [django.request:231] Internal Server Error: /servers/delete/573afe56c323c9073aa7dda8/
Traceback (most recent call last):
  File "/root/.pex/install/Django-1.7.11-py2.py3-none-any.whl.dc8612fe3bbe13c40d7a66002a1a4af1693a40e7/Django-1.7.11-py2.py3-none-any.whl/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/root/.pex/install/Django-1.7.11-py2.py3-none-any.whl.dc8612fe3bbe13c40d7a66002a1a4af1693a40e7/Django-1.7.11-py2.py3-none-any.whl/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/opt/amon/amon/apps/servers/views.py", line 83, in delete_server
    alerts_model.delete_server_alerts(server['_id'])
NameError: global name 'alerts_model' is not defined
`

**Comment by hailthemelody** [comment]
And the error from /var/log/amon_requests.log:

`
ERROR Internal Server Error: /servers/delete/573aff0ac323c9073aa7ddf0/
Traceback (most recent call last):
  File "/root/.pex/install/Django-1.7.11-py2.py3-none-any.whl.dc8612fe3bbe13c40d7a66002a1a4af1693a40e7/Django-1.7.11-py2.py3-none-any.whl/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/root/.pex/install/Django-1.7.11-py2.py3-none-any.whl.dc8612fe3bbe13c40d7a66002a1a4af1693a40e7/Django-1.7.11-py2.py3-none-any.whl/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/opt/amon/amon/apps/servers/views.py", line 83, in delete_server
    alerts_model.delete_server_alerts(server['_id'])
NameError: global name 'alerts_model' is not defined
`

**Comment by martinrusev** [comment]
@hailthemelody That is definitely a bug. Will update the code as soon as I can.

**Comment by boardyuk** [comment]
@martinrusev Has there been any movement on this as I'm also experiencing this multiple server registration issue with Ubuntu 14.04 on the hosted trial? If it's not been resolved yet is there a workaround for it?

I installed the agent via API_KEY=*removed* bash -c "$(curl -L https://*removed*.amon.cx/install/)"

**Comment by martinrusev** [comment]
@boardyuk Will look into it in a couple of hours and release a bugfix. Will ping you here when it is out

**Comment by martinrusev** [comment]
@hailthemelody, @boardyuk Trying to reproduce the bug, but I can't trigger it on my vagrant boxes. Can you please run the following:
```
amonagent -version
# The second command at least 2-3 times, to see if the machineid is the same. 
# This is what Amon uses to identify servers
amonagent -machineid 
```

**Comment by hailthemelody** [comment]
Thanks for looking into this @martinrusev , here are my results:

```bash
$ amonagent -version
Amon - Version 0.5.5
$ amonagent -machineid
baa757f55097167225f15c4307902284
$ amonagent -machineid
baa757f55097167225f15c4307902284
$ amonagent -machineid
baa757f55097167225f15c4307902284
$ amonagent -machineid
baa757f55097167225f15c4307902284
```

**Comment by boardyuk** [comment]
boardy at docker01 in [~]
10:29:23 › amonagent -version
Amon - Version 0.5.5
boardy at docker01 in [~]
10:35:15 › sudo amonagent -machineid
8b09e766f565042183e782a08855dda6
boardy at docker01 in [~]
10:36:26 › sudo amonagent -machineid
8b09e766f565042183e782a08855dda6
boardy at docker01 in [~]
10:37:11 › sudo amonagent -machineid                                               8b09e766f565042183e782a08855dda6

Strangely it seems to be working now with the hosted version when it wasn't yesterday.

**Comment by boardyuk** [comment]
After a little investigation it seems that the machine-id file does not exist when I start the service and will create duplicates in the UI, however when I run sudo amonagent -machineid it creates the machine-id file and all works as it should.

**Comment by hailthemelody** [comment]
Partially the same for me @martinrusev - I was only able to yield results with `amonagent -machineid` by running as root (`amonagent -version` ran fine without). I had assumed this was intentional and left it out in my earlier message. However, I will not have a chance to test of this makes a difference in the UI until tomorrow.

**Comment by hailthemelody** [comment]
@martinrusev , I think @boardyuk is on to something. I did a fresh install of amonagent and the `/etc/opt/amonagent/machine-id` file wasn't generated until I ran `sudo amonagent -machineid` post-install. It's been working as expected since then!

**Comment by hailthemelody** [comment]
Further, default amonagent installs don't register with the Amon service unless I append `:80` in the `/etc/opt/amonagent/amonagent.conf` file.

The following _does not_ work, which is the default file that it created:
```
{"api_key": "ABEPWqkKX7waIkApZ1hila0fFaGlb0czhrpdaxpHjbE" , "amon_instance": "http://192.168.2.43"}
```

The follow _does_ work, by adding the default port explicitly, which was not done by default:

```
{"api_key": "ABEPWqkKX7waIkApZ1hila0fFaGlb0czhrpdaxpHjbE" , "amon_instance": "http://192.168.2.43:80"}
```


 I am using the default install instructions as provided by the service, using a stock install:

```
API_KEY=ABEPWqkKX7waIkApZ1hila0fFaGlb0czhrpdaxpHjbE bash -c "$(curl -L http://192.168.2.43/install/)
```

**Comment by martinrusev** [comment]
@hailthemelody, @boardyuk Thank you so much for the thorough testing. I will make sure that`amonagent -machineid` is executed in postinst.

Will see how I can see the port issue and why it doesn't set to 80 by default

**Comment by martinrusev** [comment]
Fixed with agent version 0.6
